weave101updates-docs-1871 (#2095)
wandb/docs: weave/concepts/what-is-weave.mdx

- **Version tracking** for prompts, models, and data so you can understand what changed
- **Feedback collection** to capture human judgments and production signals

> ## The main threads of Weave

Reviewer: I think we should avoid headers immediately followed by sub-headers. Can we put an intro sentence in here that introduces what the threads of Weave are? Basically just say it contains the following feature set or whatever

### Traces

> Tracks end-to-end how a specific LLM application comes to its response.

Reviewer: ```suggestion
Track end-to-end how data flows through your LLM application.
```

"...how a specific LLM application comes to its response" 
I'm not sure we can technically say this because I don't think we really have observability into how models are arriving at their responses. I would just double check, otherwise, I would go with this suggestion or something like it

- See inputs and outputs of each application usage.
- See source documents used to produce the LLM feedback.

> - See cost, token count, latency of the LLM call.

Reviewer: ```suggestion
- See cost, token count, and latency of LLM calls.
```

- Drill down into specific prompts and how answers are produced.
- Collect feedback on responses from users.
- In your code, you can use Weave [ops and calls](/weave/guides/tracking/tracing) to track what your functions are doing.

[Get started with tracing](/weave/quickstart)

### Evaluations

> Systematic benchmarking of your application for you to know how good it is performing and be confident to deploy it to production.

Reviewer: Thoughts?
```suggestion
Systematically benchmark your LLM application's performance to ensure confidence when deploying to production."
```

imo, this is kind of an awkward sentence fragment. I think we can tighten it up into a full sentence with a clear action.

### Collect feedback

Capture human feedback, annotations, and corrections from production use. Use this data to build better test cases and improve your application.

[Collect feedback](/weave/guides/tracking/feedback)

### Monitor production

Score production traffic with the same scorers you use in evaluation. Set up guardrails to catch issues before they reach users.

[Set up guardrails and monitors](/weave/guides/evaluation/guardrails_and_monitors)

## Get started using Weave

Weave provides SDKs for Python and TypeScript. Both SDKs support tracing, evaluation, datasets, and the core Weave features. Some advanced features like class-based Models and Scorers are currently not available for the Weave TypeScript SDK.

To get started using Weave:
1. Create a Weights & Biases account at [https://wandb.ai/site](https://wandb.ai/site/?utm_source=course&utm_medium=course&utm_campaign=weave) and get your API key from [https://wandb.ai/authorize](https://wandb.ai/authorize?utm_source=course&utm_medium=course&utm_campaign=weave)
2. Install Weave:

<CodeGroup>

```Python Python
pip install weave
```

```Typescript Typescript
npm install weave
```
</CodeGroup>

> 3. In your script, import Weave and initialize a project::

Reviewer: ```suggestion
3. In your script, import Weave and initialize a project:
```

<CodeGroup>
```Python Python
import weave
client = weave.init('your-team/your-project-name')
```

```TypeScript Typescript
import * as weave from 'weave';
const client = await weave.init('your-team/your-project-name');
```
</CodeGroup>

> You're now ready to start using Weave!

Reviewer: I think we generally try to avoid this kind of statement and exclamation points 😅 . Instead, I would say something like, "Once you've configured the Weave client, you can start tracking your LLM calls," which I think segues nicely into the next step.

I also think that line somewhat contradicts the idea of a step 4.

Author: lol, so true.  admittedly I reckon this one got past me from something i copied. However, I could almost defend this particular one -- in light of Scott's emphasis on us conveying better how simple it is to get started and to hide the complexity. It doesn't, actually, contradict step 4, since if they are using an integration they really are all set.
All that being said, yep, i'll change it and clean this up

4. Weave integrates with popular LLM providers and frameworks.  When you use a [supported integration](/weave/guides/integrations/), Weave automatically traces LLM calls without additional code changes.

> However, to log traces to custom methods, add a one line decorator `weave.op` to any function. Works in development or production.

Reviewer: ```suggestion
However, you can still use Weave log traces for unsupported LLM providers by adding a one-line decorator (`weave.op`) to your call function. You can also use the decorator on any function in your app to track its input, output, and code.
```
I think this is more specific. I also am wary of using the word "method" unless I know for sure that the function is part of a class.

Author: TIL!  thanks, wasn't aware of the method/function nuance in python.
